File: mobile/app/src/main/java/com/resuera/eventhive/features/auth/AuthApi.kt
package com.resuera.eventhive.features.auth

import com.resuera.eventhive.shared.model.AuthResponse
import com.resuera.eventhive.shared.model.User
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface AuthApi {
    @POST("/api/auth/login")
    fun login(@Body user: User): Call<AuthResponse>

    @POST("/api/auth/register")
    fun register(@Body user: User): Call<AuthResponse>

    @POST("/api/auth/google-mobile")
    fun googleLogin(@Body body: Map<String, String>): Call<AuthResponse>

    @GET("/api/auth/me")
    fun getMe(): Call<AuthResponse>
}

File: web/eventhive/src/features/auth/authApi.ts
import { apiFetch } from '../../shared/api/client'
import type { UserAuth } from '../../shared/types'

export interface LoginRequest {
  email: string
  password: string
}

export interface RegisterRequest {
  firstname: string
  lastname: string
  email: string
  password: string
}

export interface AuthResponse extends UserAuth {
  status: string
}

export const authApi = {
  login: (data: LoginRequest) =>
    apiFetch<AuthResponse>('/api/auth/login', {
      method: 'POST',
      body: data,
    }),

  register: (data: RegisterRequest) =>
    apiFetch<AuthResponse>('/api/auth/register', {
      method: 'POST',
      body: data,
    }),

  getCurrentUser: () =>
    apiFetch<AuthResponse>('/api/auth/me'),
}